tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHttpMessageHandler.cs

```diff
@@ -96,6 +98,13 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage reques
                 }
             }
 
+            if (ExpectedHeaders != null)
+            {
+                Assert.IsTrue(ActualRequestMessage.Headers.Contains(TelemetryConstants.XClientLastRequest));
```

Comment:
This assertion is not strong enough. You can expose `ActualHeaders` and then validate that you're sending the correct telemetry in a variatey of use cases.

Reply:
i added more tests

---
In reply to: [325641770](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1389#discussion_r325641770) [](ancestors = 325641770)